app/models/service/aggregation.rb

```diff
@@ -9,12 +9,115 @@ module Service::Aggregation
     virtual_column :aggregate_direct_vm_disk_space_used,      :type => :integer
     virtual_column :aggregate_direct_vm_memory_on_disk,       :type => :integer
 
-    virtual_column :aggregate_all_vm_cpus,                    :type => :integer
-    virtual_column :aggregate_all_vm_memory,                  :type => :integer
-    virtual_column :aggregate_all_vm_disk_count,              :type => :integer
-    virtual_column :aggregate_all_vm_disk_space_allocated,    :type => :integer
-    virtual_column :aggregate_all_vm_disk_space_used,         :type => :integer
-    virtual_column :aggregate_all_vm_memory_on_disk,          :type => :integer
+    virtual_attribute :aggregate_all_vm_cpus,   :integer,
+                      :arel => aggregate_service_hardware_arel('cpus', :cpu_total_cores)
+    virtual_attribute :aggregate_all_vm_memory, :integer,
+                      :arel => aggregate_service_hardware_arel('memory', :memory_mb, :mb_to_bytes => true)
+    virtual_attribute :aggregate_all_vm_disk_count, :integer,
+                      :arel => aggregate_all_vms_disk_count_arel
+    virtual_attribute :aggregate_all_vm_disk_space_allocated, :integer,
+                      :arel => aggregate_all_vms_disk_space_allocated_arel
+    virtual_attribute :aggregate_all_vm_disk_space_used, :integer,
+                      :arel => aggregate_all_vms_disk_space_used_arel
+    virtual_attribute :aggregate_all_vm_memory_on_disk, :integer,
+                      :arel => aggregate_service_hardware_arel('memory_on_disk', :memory_mb, :on_vms_only => true,
+                                                                                             :mb_to_bytes => true)
+
+    AGGREGATE_ALL_VM_ATTRS = [
+      :aggregate_all_vm_cpus,
+      :aggregate_all_vm_memory,
+      :aggregate_all_vm_disk_count,
+      :aggregate_all_vm_disk_space_allocated,
+      :aggregate_all_vm_disk_space_used,
+      :aggregate_all_vm_memory_on_disk
+    ].freeze
+
+    scope :with_aggregates, -> { select(Arel.star, *AGGREGATE_ALL_VM_ATTRS) }
+  end
+
+  module ClassMethods
+    SERVICE_RESOURCES_TBL = ServiceResource.arel_table
+    VMS_TBL               = Vm.arel_table
+    HARDWARES_TBL         = Hardware.arel_table
+    DISKS_TBL             = Disk.arel_table
+    PARTITIONS_TBL        = Partition.arel_table
+
+    VM_OR_TEMPLATE_TYPE   = Arel::Nodes::SqlLiteral.new("'VmOrTemplate'")
+    ANCESTRY_MATCH        = Arel::Nodes::SqlLiteral.new("'/%'")
+    ANCESTRY_ILIKE        = Arel::Nodes::NamedFunction.new("CONCAT", [Service.arel_table[:id], ANCESTRY_MATCH])
```

Comment:
I guess I am confused about the concern here.

As this is a database backed application, and making 1 extra request to the DB to initialize this model (for each constant the has `arel_table` in it of course) doesn't seem like it would have any huge impact that we should be concerned about (I realize that latency can be a factor when an appliance is setup against a remote DB, but if it is setup where latency is so bad that it can't perform a simple table `EXPLAIN` to build the `arel_table` object on application boot, my it will fall flat when at runtime when even some of the more optimized pages... say, what I changed in this PR).  

And to me, this **IS** a constant, and shouldn't be put into a method where it isn't rebuilt and a new object is instantiated each time the `virtual_attributes` are called, in which it would generate the same exact SQL each time.  For the `/service/explorer/:id`, when `.with_aggregates` is called, that means we are instantiating `ANCESTRY_ILIKE` (well, the method/variable version of it if it were changed) 6 times for each `virtual_attribute`.

I am fine if there is idle memory concerns, stale connection issues, or the `arel_table` will change over time and cause issues (with the use of `table_alias` or something), then I would be fine changing it, but if the concern is only loading it on boot then I am not sure I follow.